db/db.go

```diff
@@ -91,3 +154,19 @@ func Stop(db *gorm.DB) error {
 	}
 	return nil
 }
+
+func IsPostgresURI(uri string) bool {
+	return strings.HasPrefix(uri, "postgresql://")
+}
+
+var txSerializer = make(chan struct{}, 1)
+
+func RunTransaction(db *gorm.DB, txFunc func(tx *gorm.DB) error) error {
```

Comment:
I saw only some instances of `db.Transaction` have been updated to use this. Is that intentional?

Reply:
This feels a bit hacky if it is done just for tests, is there a way we can isolate the changes to the test DB instead?

Reply:
I think this is ok. We can run the sqlite tests locally by default and check all of them in the GH action